_posts/2022-12-12-Remove-Unused-Code-to-Avoid-Decay.md

```diff
@@ -0,0 +1,126 @@
+    "width": 500,
+    "height": 200,
+    "config": {
+      "title":{"fontSize":14},
+      "axisY":{"titleFontSize":14, "labelFontSize":12},
+      "axisX":{"titleFontSize":14, "labelFontSize":12}
+    },
+    "data": {
+      "values": [
+        {"x": 1, "y": 28},
+        {"x": 1000, "y": 500}
+      ]
+    },
+    "mark": {"type": "line", "point": {"filled": false,
+      "fill": "white", "size":50
+    }},
+    "encoding": {
+      "x": {"title": "Lines of Code", "field": "x", "type": "quantitative", "scale": {"domain": [1, 1000]}
+      },
+      "y": {"title": "Time (minutes)", "field": "y", "type": "quantitative", "scale": {"domain": [1, 500]}}
+    }
+  }
+const embed_opt = {"mode": "vega-lite"};
+const el = document.getElementById('vis');
+const view = vegaEmbed("#vis", spec, embed_opt);
+</script>
+
+When considering multiple parts of code in many files, this maintenance can become untenable, leading to the gradual decay of your code quality or functionality. For example, let's assume one line of code costs 30 seconds to maintain (feel free to substitute time with monetary or personnel aspects as an example measure here too). 1000 lines of code would cost 500 minutes (or about 8 hours) to maintain. This becomes more complex when considering multiple files, collaborators, or languages.
+
+<i class="fas fa-hiking" style="font-size:4em;"></i>
+{:.center}
+
+Think about your project as if it were on a hiking trail: __"Carry as little as possible, but choose that little with care."__ (Earl Shaffer). Be careful what code you choose to carry; it may impact your ability to address needs over time and lead to otherwise unintended software decay.
+
+## Detecting Unused Code with Vulture
+
+Understanding the cost of added content, it's important to routinely examine which parts of your code are still necessary. You can prepare your code for a long journey by detecting (and removing) unused code with various automated tools. These tools are generally designed for static analysis and linting, meaning they may also be incorporated into automated and routine testing.
+
+```shell
+$ vulture unused_code_example.py
+unused_code_example.py:3: unused import 'os' (90% confidence)
+unused_code_example.py:4: unused import 'pd' (90% confidence)
+unused_code_example.py:7: unused function 'unused_function' (60% confidence)
+unused_code_example.py:14: unused variable 'unused_var' (60% confidence)
+```
+
+_Example of Vulture command line usage to discover unused code._
+
+[Vulture](https://github.com/jendrikseipp/vulture) is one tool dedicated to finding unused python code. Vulture provides both a command line interface and Python API for discovering unused code. It also provide a rough confidence to show how certain it was about whether the block of code was unused. See the following interactive example for a demonstration of using Vulture.
+
+[Example Link](https://cu-dbmi.github.io/notebooks/lab?path=unused_code_detection.ipynb)
```

Comment:
I'd prefer to see this link labeled by what it points to, e.g., "Unused Code Detection Example".

Reply:
Thanks, great point, I'll add this change into a forthcoming commit.